core/ArchiveProcessor/Loader.php

```diff
@@ -272,11 +271,17 @@ public function loadExistingArchiveIdFromDb()
      */
     protected function getMinTimeArchiveProcessed()
     {
-        $endDateTimestamp = self::determineIfArchivePermanent($this->params->getDateEnd());
-        if ($endDateTimestamp) {
-            // past archive
-            return $endDateTimestamp;
+        // for range periods in this request, make sure to check for the ttl
+        $isRangeArchiveAndArchivingEnabled = $this->params->getPeriod()->getLabel() == 'range'
+            && !Rules::isArchivingDisabledFor([$this->params->getSite()->getId()], $this->params->getSegment(), $this->params->getPeriod()->getLabel());
+        if (!$isRangeArchiveAndArchivingEnabled) {
+            $endDateTimestamp = self::determineIfArchivePermanent($this->params->getDateEnd());
+            if ($endDateTimestamp) {
+                // past archive
+                return $endDateTimestamp;
+            }
```

Comment:
Note: if we see a range period and we are allowed to archive and we're in a browser request, we check the ttl no matter what.

So if range archiving via browser is forced, we always archive in the browser every time the ttl is no longer valid.

If range archiving is not allowed via browser, we use the default logic (if the range does not include today, use latest archive that was archived after end date). If it includes today, we check the TTL using the existing code.